src/test/javascript/spec/component/modeling-assessment/modeling-assessment.component.spec.ts

```diff
@@ -165,6 +181,60 @@ describe('ModelingAssessmentComponent', () => {
         expect(comp.feedbacks).toEqual(mockFeedbacks);
     });
 
+    it('should filter references by result feedbacks', () => {
+        expect(comp.referencedFeedbacks).toBeEmpty();
+        expect(comp.feedbacks).toBe(undefined);
+
+        comp.umlModel = mockModel;
+        comp.resultFeedbacks = mockFeedbacks;
+
+        expect(comp.referencedFeedbacks).toEqual([mockFeedbackWithReference, mockFeedbackInvalid]);
+        expect(comp.feedbacks).toBe(mockFeedbacks);
+    });
+
+    it('should calculate drop info', () => {
+        const spy = jest.spyOn(translatePipe, 'transform');
+        comp.umlModel = mockModel;
+        comp.resultFeedbacks = [mockFeedbackWithGradingInstruction];
+
+        expect(spy).toHaveBeenCalledWith('artemisApp.assessment.messages.removeAssessmentInstructionLink');
+        expect(spy).toHaveBeenCalledWith('artemisApp.exercise.assessmentInstruction');
+        expect(spy).toHaveBeenCalledWith('artemisApp.assessment.feedbackHint');
+        expect(mockModel.assessments[0].dropInfo.instruction).toBe(mockFeedbackWithGradingInstruction.gradingInstruction);
```

Comment:
Should we assert a certain length of the ``assessments`` array before accessing the first index?

Reply:
It is a field of the `mockModel` which we create in the test preparation code so I think it is not needed.